src/app/modules/learning/learning-observer/chatbot-modal.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { LearningObserverComponent } from './learning-observer.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ChatbotModalService {
+  private readonly modalOpenInterval = 15 * 60 * 1000; //minutes * seconds * milliseconds
```

Comment:
Let's make this 20 minutes to avoid aggravating learners. Since the chatbot will activate on startup, we should be able to collect 3-4 interactions with 20 which is enough.